### PR TITLE
8306489: Open source AWT List related tests

### PR DESCRIPTION
**Pull request by kumarabhi006**
This PR is to open source List related AWT tests.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306489](https://bugs.openjdk.org/browse/JDK-8306489): Open source AWT List related tests

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13748/head:pull/13748` \
`$ git checkout pull/13748`

Update a local copy of the PR: \
`$ git checkout pull/13748` \
`$ git pull https://git.openjdk.org/jdk.git pull/13748/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13748`

View PR using the GUI difftool: \
`$ git pr show -t 13748`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13748.diff">https://git.openjdk.org/jdk/pull/13748.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13748#issuecomment-1530956034)

### CODE REVIEW

File: test/jdk/java/awt/List/ISCAfterRemoveAllTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+        if (!passed) {
+            throw new RuntimeException("Test failed.");
+        }
+    }
+
+    public void itemStateChanged(ItemEvent ie) {
+        System.out.println(ie);
+        // We shouldn't generate any events since the list is empty
+        passed = false;
```
[the remark this answers is not in the document]
Comment (kumarabhi006) [reply test/jdk/java/awt/List/ISCAfterRemoveAllTest.java:140]: Updated.

File: test/jdk/java/awt/List/ListEnterExitTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+}
+
+class MouseEnterExitListener extends MouseAdapter {
+
+    boolean passed = false;
+
+    public void mouseEntered(MouseEvent e) {
+        passed = true;
```
[the remark this answers is not in the document]
Comment (kumarabhi006) [reply test/jdk/java/awt/List/ListEnterExitTest.java]: Added separate flags to test enter and exit events.

**Review by prrace — APPROVED** [review]

**Review by mrserb — APPROVED** [review]
Looks fine, please confirm that mach5 is green.

**Pull request closed without merging**